model-optimizer/mo/back/preprocessing.py

```diff
@@ -0,0 +1,381 @@
+        num_channels_scale = len(mean_scale['scale']) if hasattr(mean_scale['scale'], '__len__') else 0
+
+        if num_channels_mean > 1 and \
+                num_channels_scale > 1 and \
+                num_channels_mean is not num_channels_scale:
+            raise Error('Mean/Scale values for {} have different sizes: {} {}'
+                        .format(ms_name, num_channels_mean, num_channels_scale))
+
+        need_guess_channels = num_channels_mean > 1 or num_channels_scale > 1
+        if not need_guess_channels:  # Mean/scale is complex and needs 'channels' specified in layout
+            continue
+
+        num_channels = num_channels_mean if num_channels_mean > 1 else num_channels_scale
+
+        for i in range(0, len(ov_function.inputs)):
+            ov_input = ov_function.input(i)
+
+            if not ov_function.get_parameters()[i].layout.empty:
+                continue
+
+            if ms_name not in ov_input.get_tensor().get_names():
+                continue
+
+            layout_item = None
+            for name in ov_input.get_tensor().get_names():
+                if name in layout_values:
+                    layout_item = layout_values[name]
+                    break
+
+            if layout_item is not None:
+                # User specified some layout, skip guessing
+                continue
+
+            if num_channels != 3:
+                raise Error('Can\'t determine channels dimension for {}. '
+                            'When number of mean/scale values is {} (not 3), '
+                            'please specify layout for input manually'.format(ms_name, num_channels))
+
+            layout_values = find_channels_dimension(shape=ov_input.get_partial_shape(),
+                                                    num_channels=num_channels,
+                                                    name=ms_name,
+                                                    layout_values=layout_values)
+    return layout_values
+
+
+def check_suitable_for_reverse(layout: Layout, ov_input):
+    """
+    Internal function. Checks if input with layout is suitable for reversing channels
+    :param: layout Existing source/target layout items specified by user
+    :param: ov_input Model's input
+    :return: True if reverse channels can be applied to input
+    """
+    if not layout_helpers.has_channels(layout):
+        return False
+    c_idx = layout_helpers.channels_idx(layout)
+    if c_idx < 0:
+        if ov_input.get_partial_shape().rank.is_dynamic():
+            return False
+        c_idx += ov_input.get_partial_shape().rank.get_length()
+    if c_idx >= ov_input.get_partial_shape().rank.get_length():
+        raise Error('Layout {} for input {} is inconsistent with shape {}'.format(
+            layout, ov_input.get_tensor().get_any_name(), ov_input.get_partial_shape()))
+    return ov_input.get_partial_shape()[c_idx] == 3 and len(ov_input.get_partial_shape()) == 4
```

Comment:
I think we shouldn't have restriction on rank.  At least for CHW (just one image) seems helpful

Reply:
Thanks, removed checking for rank == 4. Added tests for 3D (success CHW,NCHW) and 6D (failure) input cases